juju/juju
# add branch status to juju status output

## Description of change

Add branch status to the display of juju status, including listing the branches with a reference number and relative time created.  The reference number is matched up to unit's tracked by the branch.

Included are related changes to the FullStatus() call in the apiserver to gather the necessary data from the model cache.  This pr introduces the model cache into the apiserver client facade.

The model cache Branches() function is updated to be consistent with state.Branches().

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations 
1. juju bootstrap localhost tuesday
2. juju status
4.  juju deploy ubuntu -n 3
4.  juju add-branch apple
4.  juju track apple ubuntu/1
3.  juju add-branch banana
3.  juju add-branch cucumber
2.  juju status                        <- all 3 branches should be listed, ubuntu/1 should have: #1 next to it.
3.  juju status ubuntu           <- apple branch should be listed, ubuntu/1 should have: #1 next to it.
3.  juju deploy ntp ; juju add-relation ntp ubuntu
2.  juju track banana ntp/0
2.  juju status                         <- apple and banana branches should be listed, ubuntu/1 should have: #1 next to it; ntp/0 should have #2 next to it.
3.  juju status ubuntu           <- apple and banana branches should be listed, ubuntu/1 should have: #1 next to it; ntp/0 should have #2 next to it
3.  juju status ubuntu/2         <- no branches displayed
4. juju track cucumber ubuntu/2  
3.  juju track cucumber ubuntu     <- all 3 branches listed, ubuntu/2 and ubuntu/1 should have diff ref numbers

Run these juju status commands with --format yaml and --format json.

## Documentation changes

maybe
